geode-core/src/integrationTest/java/org/apache/geode/internal/net/SSLSocketHostNameVerificationIntegrationTest.java

```diff
@@ -0,0 +1,228 @@
+    this.socketCreator = SocketCreatorFactory.getSocketCreatorForComponent(CLUSTER);
+  }
+
+  @After
+  public void tearDown() throws Exception {
+    if (this.clientSocket != null) {
+      this.clientSocket.close();
+    }
+    if (this.serverSocket != null) {
+      this.serverSocket.close();
+    }
+    if (this.serverThread != null && this.serverThread.isAlive()) {
+      this.serverThread.interrupt();
+    }
+    SocketCreatorFactory.close();
+  }
+
+  @Test
+  public void nioHandshakeValidatesHostName() throws Exception {
+    ServerSocketChannel serverChannel = ServerSocketChannel.open();
+    this.serverSocket = serverChannel.socket();
+
+    InetSocketAddress addr = new InetSocketAddress(localHost, 0);
+    serverSocket.bind(addr, 10);
+    int serverPort = this.serverSocket.getLocalPort();
+
+    this.serverThread = startServerNIO(serverSocket, 15000);
+
+    await().until(() -> serverThread.isAlive());
+
+    SocketChannel clientChannel = SocketChannel.open();
+    await().until(
+        () -> clientChannel.connect(new InetSocketAddress(localHost, serverPort)));
+
+    this.clientSocket = clientChannel.socket();
+
+    SSLEngine sslEngine =
+        this.socketCreator.createSSLEngine(this.localHost.getHostName(), 1234);
+
+    try {
+      this.socketCreator.handshakeSSLSocketChannel(clientSocket.getChannel(),
+          sslEngine, 0, true,
+          ByteBuffer.allocate(500), new BufferPool(mock(DMStats.class)));
+
+      if (doEndPointIdentification && !addCertificateSAN) {
+        fail("Failed to validate hostname in certificate SAN");
+      } else {
+        assertNull(serverException);
+      }
+    } catch (SSLHandshakeException sslException) {
+      if (doEndPointIdentification && !addCertificateSAN) {
+        assertThat(sslException).hasRootCauseInstanceOf(CertificateException.class)
+            .hasStackTraceContaining("No name matching " + this.localHost.getHostName() + " found");
+      } else {
+        assertThat(sslException).doesNotHaveSameClassAs(new CertificateException(
+            "No name matching " + this.localHost.getHostName() + " found"));
+        throw sslException;
+      }
+    }
+  }
+
+  private Thread startServerNIO(final ServerSocket serverSocket, int timeoutMillis) {
+    Thread serverThread = new Thread(new MyThreadGroup(this.testName.getMethodName()), () -> {
+      NioSslEngine engine = null;
+      Socket socket = null;
+      try {
+        socket = serverSocket.accept();
+        SocketCreator sc = SocketCreatorFactory.getSocketCreatorForComponent(CLUSTER);
+        engine =
+            sc.handshakeSSLSocketChannel(socket.getChannel(),
+                sc.createSSLEngine(this.localHost.getHostName(), 1234),
+                timeoutMillis,
+                false,
+                ByteBuffer.allocate(500),
+                new BufferPool(mock(DMStats.class)));
+      } catch (Throwable throwable) {
+        throwable.printStackTrace(System.out);
```

Comment:
No need to print this stack trace

Reply:
Thanks @bschuchardt. Removed the unwanted output